pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -530,6 +1064,30 @@ def controlled_resource_decomp(
         num_ctrl_values,
         num_work_wires,
     ) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources for a controlled version of the operator.
+
+        Args:
+            outer_num_ctrl_wires (int): The number of control qubits to further control the base
+                controlled operation upon.
+            outer_num_ctrl_values (int): The subset of those control qubits, which further control
+                the base controlled operation, which are controlled when off.
```

Comment:
Can this be clearer somehow? The outer_num_ctrl_values are a subset of the control qubits? So it is a number of qubits, which are a subset of the outer-control-qubits, that control the base controlled operation, which, when "off" (what does off mean?) are controlled?

Reply:
I clarified the "when off". The reason this method is so complicated, is because it corresponds to a nested control operator. So we have multiple sets of controls. You can see that in the end we just consolidate all of the controls, values and work wires together.